lib/galaxy/datatypes/media.py

```diff
@@ -32,6 +33,94 @@ def ffprobe(path):
     return data["format"], data["streams"]
 
 
+magic_number = {
+    "mp4": {
+        "offset": 4,
+        "hex": [
+            "66 74 79 70 69",
+            "66 74 79 70 6D",
+            "66 74 79 70 4D",
```

Comment:
https://en.wikipedia.org/wiki/List_of_file_signatures 

```
66 74 79 70 69 73 6F 6D 	ftypisom 	4 	mp4 	ISO Base Media file (MPEG-4)
66 74 79 70 4D 53 4E 56 	ftypMSNV 	4 	mp4 	MPEG-4 video file 
66 74 79 70 68 65 69 63
66 74 79 70 6d 	ftypheic 	4 	heic 	High Efficiency Image Container (HEIC) 
```

6d is heic, not sure if you want to lump this in with mp4? (I assume you based this on files you had/have seen). Also might want to consider extending these magic numbers to include the whole string

Reply:
I wanted to grab m4v file format.
https://www.garykessler.net/library/file_sigs.html

> [4 byte offset]
> 66 74 79 70 6D 70 34 32 	  	[4 byte offset]
> ftypmp42
> M4V 	  	MPEG-4 video|QuickTime file

But, you are right it will be missed with the heic I will change it to  ` 66 74 79 70 6D 70 34 32 `.
Also, I will consider extending the magic numbers.